### PR TITLE
Handle solverdummy arguments properly

### LINKED ISSUE #1447: Improve solverdummy argument parsing
Our solverdummies are part of the test infrastructure. In case something goes wrong with the argument parsing, users (might) face issues like this one

https://stackoverflow.com/questions/62570882/what-basic-string-m-construct-null-not-valid

which is very cryptic and due to inappropriate handling of the parsed arguments (see the discussion in the link and our current implementation).

https://github.com/precice/precice/blob/e6da8d5d9432f0872ff0e1ecf652477fbbcf6c68/examples/solverdummies/cpp/solverdummy.cpp#L13-L17

### PR DESCRIPTION
**Pull request by davidscn**
## Main changes of this PR

See details in #1447. There was already a check for the cpp solverdummy, but it was in the wrong place. I don't think a changelog entry is necessary.

Closes #1447 

## Author's checklist

* [x] I used the [`pre-commit` hook](https://precice.org/dev-docs-dev-tooling.html#setting-up-pre-commit) to prevent dirty commits and used `pre-commit run --all` to format old commits.
* [x] I squashed / am about to squash all commits that should be seen as one.

## Reviewers' checklist

* [ ] Does the changelog entry make sense? Is it formatted correctly?
* [ ] Do you understand the code changes?

### CODE REVIEW

File: examples/solverdummies/c/solverdummy.c [hunk 1]
```diff
@@ -23,17 +23,17 @@ int main(int argc, char **argv)
   const char *writeDataName;
   const char *readDataName;
 
-  const char *configFileName  = argv[1];
-  const char *participantName = argv[2];
-
   if (argc != 3) {
-    printf("Usage: ./solverdummy configFile solverName\n\n");
+    printf("The solverdummy was called without arguments. Usage: ./solverdummy configFile solverName\n\n");
```
Comment (IshaanDesai) [inline examples/solverdummies/c/solverdummy.c:27]:
```suggestion
    printf("The solverdummy was called with incorrect arguments. Usage: ./solverdummy configFile solverName\n\n");
```

File: examples/solverdummies/cpp/solverdummy.cpp [hunk 2]
```diff
@@ -10,20 +10,20 @@ int main(int argc, char **argv)
...
 
-  std::string configFileName(argv[1]);
-  std::string solverName(argv[2]);
-  std::string meshName;
-  std::string dataWriteName;
-  std::string dataReadName;
-
   if (argc != 3) {
-    std::cout << "Usage: ./solverdummy configFile solverName\n\n";
+    std::cout << "The solverdummy was called without arguments. Usage: ./solverdummy configFile solverName\n\n";
```
Comment (IshaanDesai) [inline examples/solverdummies/cpp/solverdummy.cpp:14]:
```suggestion
    std::cout << "The solverdummy was called with incorrect arguments. Usage: ./solverdummy configFile solverName\n\n";
```

**Review by IshaanDesai — APPROVED** [review]
Looks good :+1:

File: examples/solverdummies/c/solverdummy.c (hunk 1 above)
Comment (davidscn) [reply examples/solverdummies/c/solverdummy.c:27]: It's actually the number of arguments which is checked here, so maybe I can even get more specific 
```suggestion
    printf("The solverdummy was called with an incorrect number of arguments. Usage: ./solverdummy configFile solverName\n\n");
```

**Comment by davidscn** [comment]
Parsing arguments is handled differently for fortran 
https://github.com/precice/precice/blob/3007d5f59af0555c931d2e424a1459f020861b96/examples/solverdummies/fortran/solverdummy.f90#L23-L24

I don't know if we can assert things there at all.

**Comment by uekerman** [comment]
Seems fortran does a decent job there already:

```
$ ./solverdummy
[2022-10-31 14:23:35.501966] [0x00007f26d3e65900] [error]   This participant's name is an empty string. When constructing a preCICE interface you need to pass the name of the participant as first argument to the constructor.
```

meaning `config` becomes an empty string.

Too many arguments does not lead to an error
```
$ ./solverdummy ../precice-config.xml SolverOne bla
preCICE: This is preCICE version 2.5.0
preCICE: Revision info: v2.5.0-22-ge6da8d5d9
preCICE: Build type: Debug
...
```

which I think is tolerable.

**Pull request merged**